common/pom.xml

```diff
@@ -37,6 +67,10 @@
       <scope>compile</scope> <!-- override the 'test' scope defined at parent pom.xml -->
       <optional>true</optional>
     </dependency>
+    <dependency>
+      <groupId>org.jctools</groupId>
+      <artifactId>jctools-core</artifactId>
+    </dependency>
```

Comment:
That's very cool. I only wonder how that happens inside Maven. How does maven-deploy-plugin pick up the reduced pom.xml instead of the original pom.xml even if we didn't tell anything to maven-deploy-plugin? (Maybe I'm talking to myself.)